frontend/pages/users/[slug].tsx

```diff
@@ -163,44 +297,73 @@ const ProfilePage = () => {
+                        Activity
+                    </div>
+                    <div
+                        className={`h-7 hover:cursor-pointer ${GetActiveTabClass(
+                            isActiveTab === 'Bookmarks'
+                        )}`}
+                        onClick={() => setIsActiveTab('Bookmarks')}
+                    >
+                        Bookmarks
                     </div>
                 </div>
-                <div className="mb-6 ml-14">
-                    <p className="mt-6 mb-10 text-2xl">Top Answers</p>
-                    <div className="mt-3 flex flex-col divide-y-2 divide-black border-2 border-black">
-                        {topAnswers.map((answer) => (
-                            <Activity
-                                id={answer.id}
-                                votes={answer.votes}
-                                data={answer.answer}
-                                is_answered={answer.is_answered}
-                                created_at={answer.created_at}
+                {isActiveTab === 'Activity' && (
+                    <div className="ml-24 flex w-[125%]">
+                        <div className="mb-6 mr-10">
+                            <p className="mt-6 mb-10 text-2xl">Top Questions</p>
+                            <div className="mt-3 flex flex-col divide-y-2 divide-black border-2 border-black">
+                                {topQuestions.map((question, index) => (
+                                    <Activity
+                                        key={index}
+                                        id={question.id}
+                                        votes={question.votes}
+                                        data={question.question}
+                                        is_answered={question.is_answered}
+                                        created_at={question.created_at}
+                                    />
+                                ))}
+                            </div>
+                        </div>
+                        <div className="mb-6 ml-14">
+                            <p className="mt-6 mb-10 text-2xl">Top Answers</p>
+                            <div className="mt-3 flex flex-col divide-y-2 divide-black border-2 border-black">
+                                {topAnswers.map((answer, index) => (
+                                    <Activity
+                                        key={index}
+                                        id={answer.id}
+                                        votes={answer.votes}
+                                        data={answer.answer}
+                                        is_answered={answer.is_answered}
+                                        created_at={answer.created_at}
+                                    />
+                                ))}
+                            </div>
+                        </div>
+                    </div>
+                )}
```

Comment:
Separate this by creating a render function then calling it . This is so that it can be more readable and clean

Reply:
> Separate this by creating a render function then calling it . This is so that it can be more readable and clean

Activity component/content is not part of bookmark component task